### PR TITLE
feat(feign-client): implement a http client request

### PR DESCRIPTION
**Pull request by huebner-j**
... therefore some changes had to be implemented:
(1) - The `newBornDog` UseCase was enhanced with a registration process, which requires a call to a service outside the domain 
(2) - A new infrastructure submodule, which implements a REST call to another service 
(3) - The integration is tested with WireMock

### CODE REVIEW

File: domain/src/main/java/today/stepbeyond/examples/springbootexamples/domain/DogDomainService.java [hunk 1]
```diff
@@ -7,6 +7,15 @@
 public class DogDomainService {
 
     public Dog createDog(String name) {
-        return new Dog().setName(name);
+        return new Dog()
+                .setName(name)
+                .setRegistered(false);
+    }
+
+    public Dog registerDog(Dog dog) {
```
Comment (jeskosda) [inline domain/src/main/java/today/stepbeyond/examples/springbootexamples/domain/DogDomainService.java:15]: Why cannot the Dog register himself? Why do we create and register a dog? Probably has some reasons, but dont get them 
````
dog.register();
````

File: domain/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/repository/PetRepository.java [hunk 2]
```diff
@@ -8,5 +8,7 @@ public interface PetRepository {
 
     Optional<Pet> find(UUID id);
 
-    UUID create(Pet newBornDog);
+    Pet create(Pet newBornDog);
+
+    Pet update(Pet registerDog);
```
Comment (jeskosda) [inline domain/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/repository/PetRepository.java:13]:
```suggestion
    Pet update(Pet updateDog);
```

File: infrastructure-petstore-rest-client/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/gateways/api/model/Request.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package today.stepbeyond.examples.springbootexamples.infrastructure.gateways.api.model;
+
+import java.util.UUID;
+
+public class Request {
```
Comment (jeskosda) [inline infrastructure-petstore-rest-client/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/gateways/api/model/Request.java:5]: What about a more precise naming? 
```suggestion
public class RegisterPetRequest {
```

File: domain/src/main/java/today/stepbeyond/examples/springbootexamples/domain/DogDomainService.java (hunk 1 above)
Comment (huebner-j) [reply domain/src/main/java/today/stepbeyond/examples/springbootexamples/domain/DogDomainService.java:15]: > Why cannot the Dog register himself?

This is actually what happens here. The idea is, that `DogDomainService` (part of the domain) implements the logic and the `Dog` is just the model. The business logic could also be in the `Dog` object. It is just a matter of taste, I guess.

> Why do we create and register a dog?

The idea behind this is that we have two domains. One pet domain and one pet store domain. The pet domain needs the pet store domain to register a dog. I must admit, it is a constructed example. We can talk about a more appropriate example.

File: infrastructure-petstore-rest-client/build.gradle [hunk 4]
```diff
@@ -0,0 +1,9 @@
+dependencies {
+    api 'org.springframework.cloud:spring-cloud-starter-openfeign'
+    implementation 'org.springframework:spring-context'
```
Comment (jeskosda) [inline infrastructure-petstore-rest-client/build.gradle:3]: Isn't the spring-context always included in the spring boot starters?

Comment (huebner-j) [reply infrastructure-petstore-rest-client/build.gradle]: true.

File: infrastructure-petstore-rest-client/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/gateways/api/model/Request.java (hunk 3 above)
Comment (huebner-j) [reply infrastructure-petstore-rest-client/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/gateways/api/model/Request.java:5]: thx.

File: domain/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/repository/PetRepository.java (hunk 2 above)
Comment (huebner-j) [reply domain/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/repository/PetRepository.java:13]: thx.

**Review by jeskosda — APPROVED** [review]
🚀

**Pull request merged**